changes/ticket32020

```diff
@@ -0,0 +1,4 @@
+  o Major bugfixes (onion service):
+    - Report back HS circuit failure back into the HS subsytem so we take
```

Comment:
If this fix has reachability consequences (I think it does) we might want to mention it explicitly. We've been fixing lots of reachability issues in v3 lately, and it would be nice to be able to get a quick list at some point to show our progress (and also to know if a possible bug has been fixed on master).

Reply:
Fixup: 90c21ed9fb